src/modules/events/calibration.cpp

```diff
@@ -0,0 +1,161 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2017 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include <px4_posix.h>
+#include <drivers/drv_hrt.h>
+#include <drivers/drv_led.h>
+#include <uORB/topics/calibration_status.h>
+
+#include "calibration.h"
+
+class Calibration;
+
+namespace calibration
+{
+Calibration 	*instance = nullptr;
+orb_advert_t _mavlink_log_pub = nullptr;
```

Comment:
Why keep these out of the class?

Reply:
Idea is to have only one sensor calibration running at a time and avoid the re initialization of  these variables during multiple sensor calibration instance. We can choose to make these variable as a static members of Calibration class.

Similar approach is being followed in temperature calibration.
https://github.com/PX4/Firmware/blob/master/src/modules/events/temperature_calibration/task.cpp#L62

Reply:
Advertising these again is cheap and they shouldn't be left around after a calibration is finished even if you're about to do another one. They can be private in Calibration and unadvertised() in the destructor. 

If you want you can use this uORB Publication wrapper (https://github.com/PX4/Firmware/blob/master/src/modules/uORB/Publication.hpp#L123) to handle everything for you.